src/Toml/Type/AnyValue.hs

```diff
@@ -82,8 +94,42 @@ reifyAnyValues _ []                 = Right []
 reifyAnyValues v (AnyValue av : xs) = sameValue v av >>= \Refl -> (av :) <$> reifyAnyValues v xs
 
 -- | Function for creating 'Array' from list of 'AnyValue'.
-toMArray :: [AnyValue] -> Maybe (Value 'TArray)
-toMArray [] = Just $ Array []
+toMArray :: [AnyValue] -> Either TomlBiMapError (Value 'TArray)
+toMArray [] = Right $ Array []
 toMArray (AnyValue x : xs) = case reifyAnyValues x xs of
-    Left _     -> Nothing
-    Right vals -> Just $ Array (x : vals)
+    Left _     -> leftWrongValue x
+    Right vals -> Right $ Array (x : vals)
+
+---------------------------------------------------------------------------
+-- Value mismatch error and TomlBiMapError
+---------------------------------------------------------------------------
+
+data MatchError = MatchError
+    { valueExpected :: Text
```

Comment:
Let's use `TValue` here instead of `Text` to restrict our types even further. In this field there couldn't be any other value except `TValue`